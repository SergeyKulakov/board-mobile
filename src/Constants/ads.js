import { Platform } from 'react-native'
import {
  FB_ADS_PLACEMENT_ID_IOS,
  FB_ADS_PLACEMENT_ID_ANDROID,
  ADMOB_UNIT_ID_IOS,
  ADMOB_UNIT_ID_ANDROID,
} from 'react-native-dotenv'

export const admobUnitId = Platform.select({
  // ios: 'ca-app-pub-3940256099942544/1033173712', // test unit id
  ios: ADMOB_UNIT_ID_IOS,
  android: ADMOB_UNIT_ID_ANDROID,
})

export const fbAdsPlacementId = Platform.select({
  ios: FB_ADS_PLACEMENT_ID_IOS,
  android: FB_ADS_PLACEMENT_ID_ANDROID,
})
