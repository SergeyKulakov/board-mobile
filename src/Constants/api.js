import {
  API_URL,
  SECOND_API,
  CLIENT_URL,
  APP_BUNDLE_ID,
  LOGO_PATH,
} from 'react-native-dotenv'

export const apiUrl = API_URL
export const secondApi = SECOND_API
export const clientUrl = CLIENT_URL
export const appBundleId = APP_BUNDLE_ID
export const logoPath = LOGO_PATH
