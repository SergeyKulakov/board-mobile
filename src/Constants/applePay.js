import { MERCHANT_IDENTIFIER } from 'react-native-dotenv'

const METHOD_DATA = [
  {
    supportedMethods: ['apple-pay'],
    data: {
      merchantIdentifier: MERCHANT_IDENTIFIER,
      supportedNetworks: ['visa', 'mastercard', 'amex'],
      countryCode: 'US',
      currencyCode: 'USD',
    },
  },
]

const OPTIONS = {}

export default {
  METHOD_DATA,
  OPTIONS,
}
