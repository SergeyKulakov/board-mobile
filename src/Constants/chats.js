const blockChatConst =
  '+NjSRRW1lPggeqleTXDt/HtjO4MqkwL3pNcPzkPcL7No1ih5Jm6G6dp1uzZd233JZRnjpUt4fgBwH'
const unblockChatConst =
  'pgciFniskM6vHj7TPg5eMoDe2MRe58jBBkMHKlLou9qMPZxENADgegHNpHBKAPMZbaFKL0Wbp6'

export { blockChatConst, unblockChatConst }
