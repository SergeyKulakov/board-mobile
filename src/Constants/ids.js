export const API_CALL = 'API_CALL'
export const SOCKET_EVENT = 'SOCKET_EVENT'
export const API_CALL_FAILURE = 'API_CALL_FAILURE'
