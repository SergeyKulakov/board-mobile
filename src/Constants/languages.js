export const languages = [
  {
    type: 'en',
    flag: 'english',
    name: 'English',
  },
  {
    type: 'fr',
    flag: 'french',
    name: 'French',
  },
  {
    type: 'es',
    flag: 'spanish',
    name: 'Spanish',
  },
  {
    type: 'it',
    flag: 'italian',
    name: 'Italian',
  },
  {
    type: 'pt',
    flag: 'portuguese',
    name: 'Portuguese',
  },
  {
    type: 'de',
    flag: 'deutsch',
    name: 'German',
  },
  {
    type: 'ru',
    flag: 'russian',
    name: 'Russian',
  },
  {
    type: 'hi',
    flag: 'indian',
    name: 'Hindi',
  },
  {
    type: 'bg',
    flag: 'indian',
    name: 'Bengali',
  },
  {
    type: 'te',
    flag: 'indian',
    name: 'Telugu',
  },
  {
    type: 'mr',
    flag: 'indian',
    name: 'Marathi',
  },
  {
    type: 'ta',
    flag: 'indian',
    name: 'Tamil',
  },
  {
    type: 'gu',
    flag: 'indian',
    name: 'Gujarati',
  },
  {
    type: 'kn',
    flag: 'indian',
    name: 'Kannada',
  },
  {
    type: 'ml',
    flag: 'indian',
    name: 'Malayalam',
  },
]
