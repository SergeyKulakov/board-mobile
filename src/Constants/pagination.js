export const JOBS_ELEMENTS_PER_PAGE = 20
export const SERVICE_PROVIDERS_ELEMENTS_PER_PAGE = 20
