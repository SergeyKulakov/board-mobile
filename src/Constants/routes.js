export const login = 'login'
export const initial = 'initial'
export const singUp = 'singUp'
export const forgotPass = 'forgotPass'
export const home = 'home'
export const profile = 'profile'
export const locationModal = 'modal.location'
export const servicesModal = 'modal.services'
export const sideMenu = 'sideMenu'
export const confirmSignUp = 'confirmSignUp'
export const rootLoading = 'rootLoading'
export const confirmResetPassword = 'confirmResetPassword'
export const notifications = 'notifications'
export const findJobs = 'findJobs'
export const filterJobs = 'filterJobs'
export const filterSubServices = 'modal.filterSubServices'
export const googleMapModal = 'model.googleMap'
export const verificationsPhoneOrEmail = 'modal.verificationsPhoneOrEmail'
export const jobDescription = 'jobDescription'
export const imagesSlider = 'imagesSlider'
export const postJob = 'postJob'
export const findHelp = 'findHelp'
export const currencyModal = 'modal.currency'
export const sortedModal = 'modal.sorted'
export const phonesModal = 'modal.phones'
export const serviceProviderProfile = 'serviceProviderProfile'
export const searchModal = 'model.searchModal'
export const myFavourites = 'myFavourites'
export const myJobs = 'myJobs'
export const canceledModal = 'canceledModal'
export const seePeopleWhoApplied = 'seePeopleWhoApplied'
export const selectJob = 'selectJob'
export const myRequests = 'myRequests'
export const ratingReview = 'ratingReview'
export const trackJobMap = 'trackJobMap'
export const track = 'track'
export const chats = 'chats'
export const chat = 'chat'
export const settings = 'settings'
export const deleteMyAccount = 'deleteMyAccount'
export const mapModal = 'mapModal'
export const subscriptions = 'subscriptions'
export const reviews = 'reviews'
export const report = 'report'
export const onBoarding = 'onBoarding'
