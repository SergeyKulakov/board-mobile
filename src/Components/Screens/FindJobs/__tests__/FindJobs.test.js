import React from 'react'
import { shallow } from 'enzyme'
import * as routes from 'Constants/routes'

import 'jest-styled-components'

import Component from '../FindJobs'

describe('FindJobs', () => {
  const defaultProps = {
    navigate: {
      push: () => null,
      pop: () => null,
      showSidebar: () => null,
    },
    onLoadJobsList: () => null,
    jobsPage: 0,
    jobsFilters: {},
    loadingJobId: null,
    jobsList: [],
    user: {
      username: 'userId',
      geolocation: '12.12413/32.3214123',
    },
    isEndJobsList: false,
    specificJob: null,
    onFavouriteClick: () => null,
    onLoadSpecificJob: () => null,
    onRemoveVacancy: () => null,
    loadApplyJobId: null,
    onApplyJob: () => null,
    defaultFilters: null,
    onJobShare: () => null,
    onShowPuck: () => null,
    getError: () => 'error',
    author: undefined,
    onSendError: () => null,
    isAdsLoading: false,
    componentId: 'componentId',
    t: text => text,
  }

  const serverError = {
    payload: {
      code: 'code',
    },
  }

  it('render snapshot', () => {
    const listWrapper = shallow(<Component {...defaultProps} />)

    const mapWrapper = shallow(<Component {...defaultProps} />)
    mapWrapper.setState(prevState => ({
      tabsState: { ...prevState.tabsState, index: 1 },
    }))

    const mapCardWrapper = mapWrapper.instance().renderMapCard({
      item: {
        author: 'author',
        _id: 'userId',
      },
    })

    expect([listWrapper, mapWrapper, mapCardWrapper]).toMatchSnapshot()
  })

  describe('search modal actions', () => {
    it('submit with keywords', done => {
      const props = {
        ...defaultProps,
        navigate: {
          ...defaultProps.navigate,
          showModal: jest.fn((route, { onSubmit }) => {
            setTimeout(() => {
              onSubmit({ keywords: 'keywords' })
            }, 1000)
          }),
        },
        onLoadJobsList: jest.fn(),
      }

      const wrapper = shallow(<Component {...props} />)

      wrapper.setState({
        loading: {
          pagination: false,
          jobsList: false,
        },
      })

      wrapper.instance().handleOpenSearchModal()

      expect(props.navigate.showModal).toHaveBeenCalledTimes(1)

      setTimeout(() => {
        expect(props.onLoadJobsList).toHaveBeenCalledTimes(2)
        done()
      }, 1100)
    })

    it('submit with service', done => {
      const props = {
        ...defaultProps,
        navigate: {
          ...defaultProps.navigate,
          showModal: jest.fn((route, { onSubmit }) => {
            setTimeout(() => {
              onSubmit({
                service: {
                  _id: '12',
                  title: 'title',
                },
              })
            }, 1000)
          }),
        },
        onLoadJobsList: jest.fn(),
      }

      const wrapper = shallow(<Component {...props} />)

      wrapper.setState({
        loading: {
          pagination: false,
          jobsList: false,
        },
      })

      wrapper.instance().handleOpenSearchModal()

      expect(props.navigate.showModal).toHaveBeenCalledTimes(1)

      setTimeout(() => {
        expect(props.onLoadJobsList).toHaveBeenCalledTimes(2)
        done()
      }, 1100)
    })
  })

  it('pagination', () => {
    const props = {
      ...defaultProps,
      page: 0,
      onLoadJobsList: () => null,
      isEndList: false,
      filters: {},
    }

    const wrapper = shallow(<Component {...props} />)

    wrapper.setState({
      loading: {
        pagination: false,
        jobsList: false,
      },
    })

    wrapper.instance().handleLoadJobs = jest.fn()

    wrapper.instance().handleLoadMoreJobs()

    expect(wrapper.instance().handleLoadJobs).toHaveBeenCalledTimes(1)
    expect(wrapper.instance().handleLoadJobs).toHaveBeenCalledWith({}, 1)
  })

  it('sort modal submit', done => {
    const props = {
      ...defaultProps,
      navigate: {
        showModal: jest.fn((route, { onSubmit }) => {
          setTimeout(() => {
            onSubmit()
          }, 1000)
        }),
      },
      onLoadJobsList: jest.fn(),
    }

    const wrapper = shallow(<Component {...props} />)

    wrapper.instance().handleOpenSortModal()

    wrapper.setState({
      loading: {
        pagination: false,
        jobsList: false,
      },
    })

    expect(props.navigate.showModal).toHaveBeenCalledTimes(1)
    expect(props.onLoadJobsList).toHaveBeenCalledTimes(1)

    setTimeout(() => {
      expect(props.onLoadJobsList).toHaveBeenCalledTimes(2)
      done()
    }, 1100)
  })

  it('submit set filters', done => {
    const nextFilters = {
      address: 'address',
    }
    const props = {
      ...defaultProps,
      filters: {},
      navigate: {
        push: jest.fn((route, { onSubmit }) => {
          setTimeout(() => {
            onSubmit(nextFilters)
          }, 1000)
        }),
      },
    }

    const wrapper = shallow(<Component {...props} />)

    wrapper.instance().handleLoadJobs = jest.fn()

    wrapper.instance().handleOpenFilter()

    expect(props.navigate.push).toHaveBeenCalled()

    setTimeout(() => {
      expect(wrapper.instance().handleLoadJobs).toHaveBeenCalledTimes(1)
      expect(wrapper.instance().handleLoadJobs).toHaveBeenCalledWith(
        nextFilters,
        0,
        true,
      )
      done()
    }, 1100)
  })

  describe('load jobs list', () => {
    const getProps = isSuccessful => ({
      ...defaultProps,
      onLoadJobsList: jest.fn(({ callback }) => {
        setTimeout(() => {
          callback({ error: isSuccessful ? undefined : serverError })
        }, 1000)
      }),
      onShowPuck: jest.fn(),
      getError: jest.fn(),
    })

    it('successful', done => {
      const props = getProps(true)

      const wrapper = shallow(<Component {...props} />)

      wrapper.instance().handleLoadJobs()

      expect(wrapper.state('loading')).toStrictEqual({
        jobsList: true,
        pagination: false,
      })

      expect(props.onLoadJobsList).toHaveBeenCalledTimes(1)
      expect(props.onShowPuck).not.toHaveBeenCalled()
      expect(props.getError).not.toHaveBeenCalled()

      setTimeout(() => {
        expect(wrapper.state('loading')).toStrictEqual({
          jobsList: false,
          pagination: false,
        })
        expect(props.onShowPuck).not.toHaveBeenCalled()
        expect(props.getError).not.toHaveBeenCalled()
        done()
      }, 1100)
    })

    it('failure', done => {
      const props = getProps(false)

      const wrapper = shallow(<Component {...props} />)

      wrapper.instance().handleLoadJobs()

      expect(wrapper.state('loading')).toStrictEqual({
        jobsList: true,
        pagination: false,
      })

      expect(props.onLoadJobsList).toHaveBeenCalledTimes(1)
      expect(props.onShowPuck).not.toHaveBeenCalled()
      expect(props.getError).not.toHaveBeenCalled()

      setTimeout(() => {
        expect(wrapper.state('loading')).toStrictEqual({
          jobsList: false,
          pagination: false,
        })
        expect(props.onShowPuck).toHaveBeenCalledTimes(1)
        expect(props.getError).toHaveBeenCalledTimes(1)
        expect(props.getError).toHaveBeenCalledWith(serverError)
        done()
      }, 1100)
    })
  })

  it('remove job card', () => {
    const props = {
      ...defaultProps,
      onRemoveVacancy: jest.fn(),
    }

    const wrapper = shallow(<Component {...props} />)

    expect(props.onRemoveVacancy).not.toHaveBeenCalled()

    wrapper.setState({ tabIndex: 0 })

    const job = {
      _id: 'jobId',
    }

    wrapper.instance().handleRemoveVacancyClick(job)

    expect(props.onRemoveVacancy).toHaveBeenCalledTimes(1)
    expect(props.onRemoveVacancy).toHaveBeenCalledWith('jobId')
  })

  it('open job detail', () => {
    const props = {
      ...defaultProps,
      navigate: {
        ...defaultProps.navigate,
        push: jest.fn(),
      },
    }

    const wrapper = shallow(<Component {...props} />)

    expect(props.navigate.push).not.toHaveBeenCalled()

    const job = {
      _id: 'jobId',
    }

    wrapper.instance().handleVacancyClick(job)

    expect(props.navigate.push).toHaveBeenCalledTimes(1)
    expect(props.navigate.push).toHaveBeenCalledWith(routes.jobDescription, {
      jobId: 'jobId',
    })
  })

  it('call favourite action', () => {
    const props = {
      ...defaultProps,
      onFavouriteClick: jest.fn(),
    }

    const wrapper = shallow(<Component {...props} />)

    expect(props.onFavouriteClick).not.toHaveBeenCalled()

    const job = {
      _id: 'jobId',
    }

    wrapper.instance().handleFavoriteClick(job)

    expect(props.onFavouriteClick).toHaveBeenCalledTimes(1)
    expect(props.onFavouriteClick).toHaveBeenCalledWith(job)
  })

  it('reload data', () => {
    const props = {
      ...defaultProps,
      onLoadJobsList: jest.fn(),
    }

    const wrapper = shallow(<Component {...props} />)

    wrapper
      .find('JobsList')
      .props()
      .onRefresh()

    expect(props.onLoadJobsList).toHaveBeenCalledTimes(1)
  })
})
