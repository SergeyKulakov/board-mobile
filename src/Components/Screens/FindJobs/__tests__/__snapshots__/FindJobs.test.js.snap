// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindJobs render snapshot 1`] = `
Array [
  <Styled(View)>
    <Component
      activeScreenId={0}
      isMapDataLoading={false}
      onClickBack={[Function]}
      onClickFilter={[Function]}
      onClickHamburger={[Function]}
      onClickNavItem={[Function]}
      onClickSort={[Function]}
      onSearchClick={[Function]}
      text={
        Object {
          "getTabTitle": [Function],
          "title": "Find Jobs",
        }
      }
    />
    <JobsList
      data={Array []}
      isAdsLoading={false}
      isPaginationLoading={false}
      isRefreshing={true}
      loadingApplyId={null}
      loadingFavoriteId={null}
      onApplyClick={[Function]}
      onFavoriteClick={[Function]}
      onLoadMoreVacancies={[Function]}
      onRefresh={[Function]}
      onRemoveVacancy={[Function]}
      onShareVacancyClick={[Function]}
      onVacancyClick={[Function]}
    />
  </Styled(View)>,
  <Styled(View)>
    <Component
      activeScreenId={1}
      isMapDataLoading={true}
      onClickBack={[Function]}
      onClickFilter={[Function]}
      onClickHamburger={[Function]}
      onClickNavItem={[Function]}
      onClickSort={[Function]}
      onSearchClick={[Function]}
      text={
        Object {
          "getTabTitle": [Function],
          "title": "Find Jobs",
        }
      }
    />
    <Map
      data={Array []}
      extraData={
        Array [
          null,
        ]
      }
      onLoadMore={[Function]}
      renderMapCard={[Function]}
      user={
        Object {
          "geolocation": "12.12413/32.3214123",
          "username": "userId",
        }
      }
    />
  </Styled(View)>,
  <ForwardRef
    onPress={[Function]}
  >
    <UNDEFINED
      _id="userId"
      author="author"
      isFavouriteLoading={false}
      onButtonClick={[Function]}
      onFavouriteClick={[Function]}
      onRemoveClick={[Function]}
      onShareClick={[Function]}
    />
  </ForwardRef>,
]
`;
